torch/onnx/verification.py

```diff
@@ -122,10 +122,14 @@ def _compare_ort_pytorch_outputs(
     atol: float,
     check_shape: bool,
     check_dtype: bool,
+    flatten: bool,
```

Comment:
You'll probably need to rebase now that #82622 is in. Please consider appending to docstring explaining what `flatten` does. Because it is not very clear to me as there is also some sort of flattening being done when `flatten==False`.

Reply:
`flatten` here actually stands for `tracing`(True) or `scripting`(False), and it it's tracing(True), we use torch.jit._flatten which ignores None type in torch output, but scripting(False) uses `_inline_flatten_list` which unpacks it and keep None type, and we need None type for Optional type tests.

It is a tricky part that `flatten` actually comes from https://github.com/pytorch/pytorch/blob/bb1e3d8008fc0e40133a3b08bcfb598cd084897b/test/onnx/onnx_test_common.py#L99-L117

I found `flatten` was decided by the model is traced or scripted, and this is exactly what I need, so I extend it all the way down to output processing. 

PLEASE ADVICE: This could be a really bad idea reusing a parameter that serves as other purpose, as it's mixing up information, but otherwise, I will have to add another parameter and pass it in from the very beginning.

Reply:
I think reusing the flatten parameter might be a bit confusing in this scenario, can we decide if a model is trace/script based on module type?

Reply:
Maybe name it `ignoreNone` or something similar?

Reply:
Done.

Reply:
Done.

Reply:
I committed to the wrong PR, so I messed up this PR with `Move Optional Type to runtime` ... They are now squashed together... Please have another review on this Thanks!